Plugins/JiraCommitHintPlugin/JiraCommitHintPlugin.cs

```diff
@@ -17,13 +18,14 @@ public class JiraCommitHintPlugin : GitPluginBase, IGitPluginForRepository
         private Jira jira;
         private string query;
         private string stringTemplate = defaultFormat;
-        private readonly BoolSetting enabledSettings = new BoolSetting("Jira hint plugin enabled", false);
-        private readonly StringSetting urlSettings = new StringSetting("Jira URL", @"https://jira.atlassian.com");
-        private readonly StringSetting userSettings = new StringSetting("Jira user", string.Empty);
-        private readonly PasswordSetting passwordSettings = new PasswordSetting("Jira password", string.Empty);
-        private readonly StringSetting jdlQuerySettings = new StringSetting("JDL Query", "assignee = currentUser() and resolution is EMPTY ORDER BY updatedDate DESC");
-        private readonly StringSetting stringTemplateSetting = new StringSetting("Jira Message Template", "Message Template", defaultFormat);
-        private readonly StringSetting jiraFields = new StringSetting("Jira fields", $"{{{string.Join("} {", typeof(Issue).GetProperties().Where(i => i.CanRead).Select(i => i.Name).OrderBy(i => i).ToArray())}}}");
+        private readonly BoolSetting enabledSettings = new BoolSetting("Jira hint plugin enabled", new TranslationString("Jira hint plugin enabled").Text, false);
+        private readonly StringSetting urlSettings = new StringSetting("Jira URL", new TranslationString("Jira URL").Text, @"https://jira.atlassian.com");
+        private readonly StringSetting userSettings = new StringSetting("Jira user", new TranslationString("Jira user").Text, string.Empty);
+        private readonly PasswordSetting passwordSettings = new PasswordSetting("Jira password", new TranslationString("Jira password").Text, string.Empty);
+        private readonly StringSetting jdlQuerySettings = new StringSetting("JDL Query", new TranslationString("JQL Query").Text, "assignee = currentUser() and resolution is EMPTY ORDER BY updatedDate DESC");
```

Comment:
I have kept the key name "JDL Query" instead of fixing it ( that's "JQL").
@RussKie @ierof Do you think I should rename the key with the effect to being unable to load the query setting (once the plugin will be released)?

Or perhaps do a transition release containing the loading of the 2 settings, keep the good one and save it...

Reply:
I don't think I fully understand the context, sorry.
Are you referring to the translation key or something else?

Reply:
That's just that before the used of the translation strings, the same string was used as the key to store the setting and the label. I fixed the typo when introducing the translation string (now that the 2 uses are now uncorrelated) so the setting key still contains the typo.

I just if you think that I should also fix the setting key name (and loose the config in the upgrade) or keep the name with the typo and that will stay strange.

Reply:
Ahhh right, I just noticed - `new StringSetting("JDL Query", new TranslationString("JQL Query")`. Thank you.

I suppose, having released this into the wild, we need to keep the existing setting. Removing it will likely generate some noise. 
If we want to rename it (to correct the typo) we need to do a fallback load, something like (pseudocode):

```
var jqlQuerySettings = new StringSetting("JQL Query"...) ?? new StringSetting("JDL Query", ...)
```

In either case, to reduce wtf, it is a good idea to leave a comment explaining why we are doing this strange things.

Reply:
Need to rename the variable too btw :)

Reply:
@RussKie That's not as easy as your pseudocode. I also thought it will be but in fact, no....
Because here, this is just the declaration of the setting and the value is not read here (but later, elsewhere with a setting source), so you can't know if the new setting will contains a value...

I don't know how to solve that.
And i'm not sure it worth the investment...

I think I will just add a comment. What do you think?